contracts/voting-verifier/src/execute.rs

```diff
@@ -142,7 +141,9 @@ pub fn end_poll(deps: DepsMut, env: Env, poll_id: PollID) -> Result<Response, Co
         .collect::<Vec<&Message>>();
 
     for message in messages {
-        VERIFIED_MESSAGES.save(deps.storage, &message.id, message)?;
+        if !is_message_verified(deps.as_ref(), message)? {
+            VERIFIED_MESSAGES.save(deps.storage, &message.id, message)?;
+        }
```

Comment:
One message mismatch will abort the whole batch. I think it's ok though, if this actually happens we will need to intervene and migrate things.